presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java

```diff
@@ -911,4 +913,42 @@ public static boolean startsWith(@SqlType("varchar(x)") Slice source, @SqlType("
         }
         return source.compareTo(0, prefix.length(), prefix, 0, prefix.length()) == 0;
     }
+
+    @Description("Translate characters from the source string based on original and translations strings")
+    @ScalarFunction
+    @LiteralParameters({"x", "y", "z"})
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice translate(@SqlType("varchar(x)") Slice source, @SqlType("varchar(y)") Slice originals, @SqlType("varchar(z)") Slice translations)
+    {
+        int[] translationsCodePoints = castToCodePoints(translations);
+        int translationsLength = translationsCodePoints.length;
+        int[] originalsCodePoints = castToCodePoints(originals);
+        if (originalsCodePoints.length != translationsLength) {
```

Comment:
This could use `checkCondition` from `Failures`. We can also rephrase the error message a bit:
```java
checkCondition(originalsCodePoints.length == translationsLength, INVALID_FUNCTION_ARGUMENT, "Translate originals and translations must be the same length");

Reply:
You mentioned that previously but I spaced - - thanks; added.